src/actions/profileAction.js

```diff
@@ -0,0 +1,32 @@
+import axios from "axios";
+import ajax from "../utils/ajax";
+import {GET_PROFILE_PENDING, GET_PROFILE_FULFILLED, GET_PROFILE_REJECTED} from "../constants/actionTypes";
+
+export const getprofilepending = () => async dispatch => {
+  await dispatch({type: GET_PROFILE_PENDING});
+};
+
+export const getprofile = uniqueCode => async dispatch => {
+  try {
+    const url = `${process.env.REACT_APP_BASE_URL}/user/${uniqueCode}`;
+    const res = await axios.get(url);
+    const response = res.response.data;
+
+    await dispatch({
+      type: GET_PROFILE_FULFILLED,
+      payload: {
```

Comment:
You know you can just pass `response.data` to payload?
`payload: response.data`
And for profile URL.
you can just do this.
```
const profileUrl = `${process.env.REACT_APP_PROFILE_URL}${uniqueCode}`;
const moldResponse = Object.assign({}, {profileUrl}, response.data);
 //console.log(moldResponse)
...
payload: moldResponse
```